thewalrus/samples.py

```diff
@@ -117,7 +117,7 @@ def generate_hafnian_sample(
         local_mu = np.zeros(2 * N)
     else:
         local_mu = mean
-    A = Amat(Qmat(cov), hbar=hbar)
+    A = Amat(cov, hbar=hbar)
```

Comment:
I actually think this line can be removed altogether, since `A` will only be used if `approx` and, in that case, another `A` will always be created on line 129 below replacing this `A` anyway.
```suggestion
```